File: implementations/flutter/design_patterns_flutter/lib/patterns/structural/decorator/constants/add_on.dart
import '../cheese_decorator.dart';
import '../bacon_decorator.dart';
import '../double_patty_decorator.dart';
import '../burger.dart';

class AddOnMeta {
  final String label;
  final Burger Function(Burger) build;

  const AddOnMeta({required this.label, required this.build});
}

/// Keys: 'cheese', 'bacon', 'double'
final Map<String, AddOnMeta> ADDONS = {
  'cheese': AddOnMeta(
    label: 'Queso (+\$0.75)',
    build: (b) => CheeseDecorator(b),
  ),
  'bacon': AddOnMeta(
    label: 'Tocineta (+\$1.10)',
    build: (b) => BaconDecorator(b),
  ),
  'double': AddOnMeta(
    label: 'Extra carne (+\$2.20)',
    build: (b) => DoublePattyDecorator(b),
  ),
};

File: implementations/react-native/design-patterns-rn/src/patterns/structural/decorator/constants/add-on.ts
import { Bacon } from '../concrete-bacon-decorator';
import { Cheese } from '../concrete-cheese-decorator';
import { DoublePatty } from '../concrete-double-decorator';
import type { Burger } from '../interface-component';
import type { BurgerDecorator } from '../interface-decorator';
import type { AddOnKey } from '../types/add-on';

export const ADDONS: Record<
  AddOnKey,
  { label: string; build: (b: Burger) => BurgerDecorator }
> = {
  cheese: { label: 'Queso (+$0.75)', build: (b) => new Cheese(b) },
  bacon:  { label: 'Tocineta (+$1.10)',  build: (b) => new Bacon(b)  },
  double: { label: 'Extra carne (+$2.20)', build: (b) => new DoublePatty(b) },
};
